util/device_profile_data.py

```diff
@@ -33,32 +33,33 @@ def extract_profile_data(device_output_file):
         ValueError: If LLVM profile data cannot be detected in the device
             output or its length or checksum is incorrect.
     """
-    lines = device_output_file.read().decode('utf-8', 'ignore').splitlines()
-    for i, line in zip(reversed(range(len(lines))), reversed(lines)):
-        match = re.match(
-            r"""
-                LLVM\ profile\ data
-                \ \(length:\ (?P<length>\d*),
-                \ CRC32:\ (?P<checksum>[0-9A-F]*)\):
-            """, line, re.VERBOSE)
-        if match:
-            exp_length = int(match.group('length'))
-            exp_checksum = match.group('checksum')
-            byte_array = bytes.fromhex(lines[i + 1])
-            break
-    # Check if output has LLVM profile data
+    device_output = re.sub(r'^.*] (.*)$',
+                           r'\1',
+                           device_output_file.read().decode('ascii', 'ignore'),
+                           flags=re.MULTILINE)
+    device_output = device_output.translate(
+        device_output.maketrans('', '', '\r\n'))
+    match = re.search(
+        r"""
+            LLVM\ profile\ data\ \(length:\ (?P<len>\d+)\ bytes,\ CRC32:\ (?P<crc>0x[0-9a-f]*)\):
+            (?P<data> 0x [0-9a-f]+)
+            \x04
+        """, device_output, re.VERBOSE)
```

Comment:
No because we remove the new lines in the line just above this one for a simpler regexp.

There is also [this](https://cs.opensource.google/opentitan/opentitan/+/master:sw/host/opentitanlib/src/uart/console.rs;drc=3cc6786654a3bb4e5f6e81dc45b7c9aee34515eb;l=90).